common/lib/client/auth.js

```diff
@@ -38,27 +38,25 @@ var Auth = (function() {
 		return JSON.stringify(c14nCapability);
 	}
 
+	function containsAuthInfo(options) {
+		var authMethods = ['authUrl', 'authCallback', 'token', 'tokenDetails', 'key'];
+		return (Utils.arrAny(authMethods, function(method) {
+			return (method in options);
+		}));
+	}
+
 	function Auth(client, options) {
 		this.client = client;
 		this.tokenParams = options.defaultTokenParams || {};
 
-		/* RSA7a4: if options.clientId is provided and is not
-		 * null, it overrides defaultTokenParams.clientId */
-		if(options.clientId) {
-			this.tokenParams.clientId = options.clientId;
-			this.clientId = options.clientId
-		}
-
 		/* decide default auth method */
 		var key = options.key;
 		if(key) {
 			if(!options.clientId && !options.useTokenAuth) {
```

Comment:
> I'm not sure I agree with this bit - should merely providing defaultTokenParams really force token auth used if it wouldn't otherwise have been used?

True, you are right